File: src/main/java/dev/paigewatson/layoutmaster/models/goods/ProducedGood.java
package dev.paigewatson.layoutmaster.models.goods;

public class ProducedGood implements DeliverableGood
{
    private final GoodsType goodsType;
    private final boolean isAssigned;

    public ProducedGood(GoodsType goodsType)
    {
        this(goodsType, false);
    }

    public ProducedGood(GoodsType goodsType, boolean isAssigned)
    {
        this.goodsType = goodsType;
        this.isAssigned = isAssigned;
    }

    public boolean isOfType(GoodsType expectedType)
    {
        return expectedType == this.goodsType;
    }

    public boolean isAssigned()
    {
        return isAssigned;
    }
}

File: src/main/java/dev/paigewatson/layoutmaster/models/goods/ProducedGood.kt
package dev.paigewatson.layoutmaster.models.goods

class ProducedGood @JvmOverloads constructor(private val goodsType: GoodsType, val isAssigned: Boolean = false) :
    DeliverableGood {
    fun isOfType(expectedType: GoodsType): Boolean {
        return expectedType === goodsType
    }

    override fun toString(): String {
        return "ProducedGood{" +
                "goodsType=" + goodsType +
                ", isAssigned=" + isAssigned +
                '}'
    }
}
